test/retry_test.go

```diff
@@ -74,12 +78,31 @@ spec:
 		t.Fatalf("Failed to get PipelineRun %q: %v", pipelineRunName, err)
 	}
 
-	// PipelineRunStatus should have 1 TaskRun status, and it should be failed.
-	if len(pr.Status.TaskRuns) != 1 {
-		t.Errorf("Got %d TaskRun statuses, wanted %d", len(pr.Status.TaskRuns), numRetries)
-	}
-	for taskRunName, trs := range pr.Status.TaskRuns {
-		if !isFailed(t, taskRunName, trs.Status.Conditions) {
+	if embeddedStatusValue != config.MinimalEmbeddedStatus {
+		// PipelineRunStatus should have 1 TaskRun status, and it should be failed.
+		if len(pr.Status.TaskRuns) != 1 {
+			t.Errorf("Got %d TaskRun statuses, wanted %d", len(pr.Status.TaskRuns), numRetries)
+		}
+		for taskRunName, trs := range pr.Status.TaskRuns {
+			if !isFailed(t, taskRunName, trs.Status.Conditions) {
+				t.Errorf("TaskRun status %q is not failed", taskRunName)
+			}
+		}
+	} else {
```

Comment:
I think this should be `if embeddedStatusValue != config.FullEmbeddedStatus` so that you're testing that the correct value is present for `both`. also "embeddedstatusvalue" is a bit redundant-- could just be "embeddedstatus".